src/uu/ls/src/ls.rs

```diff
@@ -1366,8 +1366,16 @@ fn list(locs: Vec<&Path>, config: Config) -> UResult<()> {
 
     display_items(&files, &config, &mut out);
 
-    for dir in &dirs {
-        enter_directory(dir, &config, initial_locs_len, &mut out);
+    for (pos, dir) in dirs.iter().enumerate() {
+        // Print dir heading - name... 'total' comes after error display
+        if initial_locs_len > 1 || config.recursive {
+            if pos.eq(&0usize) && files.is_empty() {
+                let _ = writeln!(out, "{}:", dir.p_buf.display());
```

Comment:
why let _ ?

Reply:
You're right -- writeln! doesn't return a Result like write!  I'll change.

Reply:
Oops, no, that's not right -- and there is a lint for this.  Do you mean why not propagate an error?  Hmm, we don't anywhere else in 'ls' yet, and that would seem to be something we may want to do all at once?

Reply:
what is the error ?

Reply:
The problem here is that `println!` just panics when it can't write to stdout but `writeln!` returns a result. And the result is then marked as `must_use`, requiring `let _ =` In that light, I think it's fine to just discard the error (or maybe unwrap?). We mostly do the same in other utils and in my opinion it's not really worth it to keep track of these errors.